### ISSUE TITLE
[Bug] Profiler displays "Logged as" with bad email if trying to change user email with another

### ISSUE THREAD

**Issue by bastien70** [symfony/symfony#59335]
### Symfony version(s) affected

7.1.5

### Description

In the profiler, I am connected as admin@admin.fr.

![image](https://github.com/user-attachments/assets/5eb04862-66b6-4cf1-978a-f972f9d89e32)

If I go to my profile, on a form allowing me to change some of my information (email, last name, first name, etc.), and I enter an already existing email even though I have a uniqueness constraint on the email, I will get the error indicating that the email has already been taken. However, looking at the profiler we will see that "Logged as" displays the email I am trying to spoof

![image](https://github.com/user-attachments/assets/d24761ee-10e8-46e7-803f-1159828dd1f2)

### How to reproduce

All you need is a form allowing you to update your email address, while having a UniqueEntity constraint on the "email" field of your "User" entity.

### Additional Context

Reproducer : https://github.com/bastien70/profiler_bad_email_reproducer

**Comment by chalasr** [comment]
We would need a reproducer to confirm such a complex bug. Can you provide one?

**Comment by bastien70** [comment]
> We would need a reproducer to confirm such a complex bug. Can you provide one?

Yes, there is a reproducer : https://github.com/bastien70/profiler_bad_email_reproducer

**Comment by MatTheCat** [comment]
This is expected: even if the email change is not persisted in the database it will stay in memory while the request is being handled.

**Comment by smnandre** [comment]
> All you need is a form allowing you to update your email address

Just a quick note: it's strongly recommended to avoid directly changing the _identity_ of an authenticated user without a full refresh (cookies, sessions, etc.) More generally, changing an email or username must be done with extreme caution..
-> great OWASP article on that topic: https://owasp.org/www-community/pages/controls/Changing_Registered_Email_Address_For_An_Account

**Comment by xabbuh** [comment]
closing as explained in https://github.com/symfony/symfony/issues/59335#issuecomment-2566654901
